strawberry/chalice/views.py

```diff
@@ -0,0 +1,153 @@
+import functools
+from http import HTTPStatus
+
+from chalice.app import CaseInsensitiveMapping, Request, Response
+from strawberry.chalice.graphiql import render_graphiql_page
+from strawberry.http import GraphQLHTTPResponse, process_result
+from strawberry.schema import BaseSchema
+from strawberry.types import ExecutionResult
+
+
+class GraphQLView:
+    def __init__(self, schema: BaseSchema, render_graphiql=True):
+        self._schema: BaseSchema = schema
+        self.graphiql: bool = render_graphiql
+
+    @staticmethod
+    @functools.lru_cache()
+    def render_graphiql() -> str:
+        """
+        Returns a string containing the html for the graphiql webpage. It also caches the
+        result using lru cache. This saves loading from disk each time it is invoked.
+        Returns:
+
+        """
+        result = render_graphiql_page()
+        return result
+
+    @staticmethod
+    def has_html_been_asked_for(headers: CaseInsensitiveMapping) -> bool:
+        """
+        Do the headers indicate that the invoker has requested html?
+        Args:
+            headers: A dictionary containing the headers in the request
+
+        Returns:
+            Whether html has been requested True for yes, False for no
+        """
+        try:
+            accept_headers = headers.get("accept")
+        except (KeyError, TypeError):
+            return False
```

Comment:
Will you get a `KeyError` with `.get()`?

Reply:
or a `TypeError` for that matter

Reply:
You're right no key error - type error maybe people can do weird stuff when they make requests.

Reply:
I've removed the keyerror, but left the typeerror for now

Reply:
I'm trying to work out if it's possible to pass headers in (not as a caseinsensitivemapping) it certainly is for the application json content, so I've left the TypeError in to be safe.